addons/mass_mailing/models/mailing_contact.py

```diff
@@ -187,7 +187,12 @@ def _message_get_default_recipients(self):
 
     def action_import(self):
         action = self.env["ir.actions.actions"]._for_xml_id("mass_mailing.mailing_contact_import_action")
-        action['context'] = self.env.context
+        context = self.env.context.copy()
+        action['context'] = context
+        if (not context.get('default_mailing_list_ids', False) and context.get('from_mailing_list', False) and context.get('previous_active_ids')):
```

Comment:
Don't think you need those "False" default values, it will return "None" without it, which is also Falsy.